hyundai_kia_connect_api/KiaUvoApiEU.py

```diff
@@ -275,6 +291,178 @@ def login(self, username: str, password: str) -> Token:
         )
 
+    def _get_authorization_code_genesis(self, username: str, password: str) -> str:
+        """
+        Authentication flow for Genesis brand.
+        """
+        _LOGGER.debug(f"{DOMAIN} - Using Genesis authentication flow")
+
+        # Create a session for handling cookies
+        session = requests.Session()
+        session.headers.update(
+            {
+                "User-Agent": USER_AGENT_GENESIS,
+                "Accept-Language": f"{self.LANGUAGE}-{self.LANGUAGE.upper()},{self.LANGUAGE};q=0.9",
+            }
+        )
+
+        # Step 1: Get initial cookie
+        url = f"https://{self.BASE_DOMAIN}/api/v1/user/oauth2/authorize?response_type=code&client_id={self.CCSP_SERVICE_ID}&redirect_uri=https://{self.LOGIN_FORM_HOST}/realms/eugenesisidm/ga-api/redirect2&lang={self.LANGUAGE}&scope=url.newapp"
+        response = session.get(url, allow_redirects=False)
+        _LOGGER.debug(
+            f"{DOMAIN} - Initial OAuth2 request status: {response.status_code}"
+        )
+        _LOGGER.debug(f"{DOMAIN} - Initial OAuth2 URL: {url}")
+
+        if response.status_code != 302:
+            _LOGGER.error(
+                f"{DOMAIN} - Initial OAuth2 request failed with status: {response.status_code}"
+            )
+            raise AuthenticationError("Failed to get initial OAuth2 redirect")
+
+        # Step 2: Follow redirect to authorize page
+        location_url = response.headers["Location"]
+        _LOGGER.debug(f"{DOMAIN} - Redirect location: {location_url}")
+        response = session.get(location_url)
+        _LOGGER.debug(
+            f"{DOMAIN} - Authorize page request status: {response.status_code}"
+        )
+
+        if response.status_code != 200:
+            _LOGGER.error(
+                f"{DOMAIN} - Failed to load authorize page with status: {response.status_code}"
+            )
+            raise AuthenticationError("Failed to load authorize page")
+
+        # Step 3: Get session
+        url = f"https://{self.BASE_DOMAIN}/api/v1/user/session"
+        response = session.get(url)
+        _LOGGER.debug(f"{DOMAIN} - Session request status: {response.status_code}")
+
+        if response.status_code != 204:
+            _LOGGER.error(
+                f"{DOMAIN} - Failed to establish session with status: {response.status_code}"
+            )
+            raise AuthenticationError("Failed to establish session")
+
+        # Step 4: Set language
+        url = f"https://{self.BASE_DOMAIN}/api/v1/user/language"
+        headers = {"Content-Type": "text/plain;charset=UTF-8"}
+        response = session.post(url, headers=headers, json={"lang": self.LANGUAGE})
+        _LOGGER.debug(f"{DOMAIN} - Set language request status: {response.status_code}")
+
+        if response.status_code != 204:
+            _LOGGER.error(
+                f"{DOMAIN} - Failed to set language with status: {response.status_code}"
+            )
+            raise AuthenticationError("Failed to set language")
```

Comment:
This looks to be duplicate of _set_session_language so that should be used for supportability.